cirq/google/engine/engine.py

```diff
@@ -388,6 +395,31 @@ def _infer_gcs_results(self, job_config: JobConfig) -> None:
             job_config.gcs_results = '{}jobs/{}'.format(job_config.gcs_prefix,
                                                         job_config.job_id)
 
+    def _make_request(self, request: HttpRequest) -> Dict:
+        RETRYABLE_ERROR_CODES = [500, 503]
+        current_delay = 0.1  #100ms
+
+        while True:
+            try:
+                return request.execute()
+            except ConnectionResetError:
+                message = "Lost connection to the engine."
+            except HttpError as rawErr:
+                err = json.loads(rawErr.content).get('error')
+                message = err.get('message')
+                # Raise RuntimeError for exceptions that are not retryable.
+                # Otherwise, pass through to retry.
+                if not err.get('code') in RETRYABLE_ERROR_CODES:
+                    raise RuntimeError(message) from None
```

Comment:
Why `from None` instead of `from rawErr`?

Reply:
I was avoiding the "another exception occurred while handling..." messaging. Agreed that raising from rawErr is useful and updated.